From a85d0ccbcd34d7869cf8e8972ca07a4b3605b29e Mon Sep 17 00:00:00 2001
From: blobfish <blobfish@gmx.com>
Date: Tue, 29 Sep 2020 06:41:32 -0400
Subject: [PATCH 03/18] BRepFill_Filling: WireFromList: We can't assume that a
 connected edge was found and the iterator is valid or we will crash on
 Edges.Remove

---
 src/BRepFill/BRepFill_Filling.cxx | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/BRepFill/BRepFill_Filling.cxx b/src/BRepFill/BRepFill_Filling.cxx
index 6266e1ba32..dfa6c56ad6 100644
--- a/src/BRepFill/BRepFill_Filling.cxx
+++ b/src/BRepFill/BRepFill_Filling.cxx
@@ -95,6 +95,7 @@ static TopoDS_Wire WireFromList(TopTools_ListOfShape& Edges)
   while (!Edges.IsEmpty())
   {
     TopTools_ListIteratorOfListOfShape itl(Edges);
+    bool found = false;
     for (; itl.More(); itl.Next())
     {
       anEdge = TopoDS::Edge(itl.Value());
@@ -117,11 +118,17 @@ static TopoDS_Wire WireFromList(TopTools_ListOfShape& Edges)
           anEdge.Reverse();
           V2 = V3;
         }
+        found = true;
         break;
       }
     }
-    BB.Add(aWire, anEdge);
-    Edges.Remove(itl);
+    if (found)
+    {
+      BB.Add(aWire, anEdge);
+      Edges.Remove(itl);
+    }
+    else
+      break;
   }
 
   aWire.Closed(Standard_True);
-- 
2.45.0

